agent/hcp/client/client.go

```diff
@@ -120,6 +143,31 @@ func (c *hcpClient) PushServerStatus(ctx context.Context, s *ServerStatus) error
 	return err
 }
 
+func (c *hcpClient) InitMetricsClient(ctx context.Context, endpoint string) error {
+	hcpConfig, err := c.cfg.HCPConfig()
+	if err != nil {
+		return err
+	}
+
+	// TODO:In follow up PR (CC-4635) : Need to use oauth2 transport in the otlpmetrichttp client to add token.
+	// We likely need to fork and update the otlpmetrichttp client, as there is no interface to do this currently.
+	exp, err := otlpmetrichttp.New(ctx, otlpmetrichttp.WithEndpoint(endpoint), otlpmetrichttp.WithTLSClientConfig(hcpConfig.APITLSConfig()))
```

Comment:
Note for reviewers: I initially wanted to have this directly in the Exporter (telemetry/exporter.go), however, to do HCP auth, we need the `hcpConfig`.

After looking at different options, this seemed like the best as the HCP client.... is a client.
The exporter is depending on a client to do the actual export. 

With this design, it made mocking very easy, and so testing extremely easy as well.